includes/gateway_request_handlers.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/**
+ * Handle IPN/webhook requests from gateways
+ * notifying site of subscription cancellation.
+ *
+ * @since TBD
+ *
+ * @param string $subscription_transaction_id The subscription transaction ID.
+ * @param string $gateway The gateway that sent the request.
+ * @param string $gateway_environment 'live' or 'sandbox'.
+ *
+ * @return string Entry to add to IPN/webhook log.
+ */
+function pmpro_handle_subscription_cancellation_at_gateway( $subscription_transaction_id, $gateway, $gateway_environment ) {
+	// Find subscription.
+	$subscription = PMPro_Subscription::get_subscription_from_subscription_transaction_id( $subscription_transaction_id, $gateway, $gateway_environment );
+	if ( empty( $subscription ) ) {
+		// The subscription does not exist on this site. Bail.
+		return 'ERROR: Could not find this subscription to cancel (subscription_transaction_id=' . $subscription_transaction_id . ').';
```

Comment:
Whenever we return strings it's good to wrap them in i18n WP functions (unless we rely on the error text being in English always).

Reply:
Discussed in Slack, logs should be in English (at least for now).